File: atcoder/abc142A_odds_of_oddness.hs
-- unihernandez22
-- https://atcoder.jp/contests/abc142/tasks/abc142_a
-- basic probability

main = do
  li <- getLine
  let n  = (read li)
  putStrLn.show $
    (fromIntegral $ floor $ (n + 1) / 2) / n

File: atcoder/abc142A_odds_of_oddness.ml
(* vicfred
 * https://atcoder.jp/contests/abc142/tasks/abc142_a
 * math
 * *)
Scanf.scanf "%d" @@ fun n ->
    Printf.printf "%.9f\n" (float_of_int (n - n / 2) /. float_of_int n)

